File: OpenTokFs.Json/RequestTypes/OpenTokVideoLayout.cs
using Newtonsoft.Json;

namespace OpenTokFs.Json.RequestTypes {
    public class OpenTokVideoLayout {
        [JsonProperty("type")]
        public string Type { get; set; }

        [JsonProperty("stylesheet", NullValueHandling = NullValueHandling.Ignore)]
        public string Stylesheet { get; set; }

        public static OpenTokVideoLayout BestFit => new OpenTokVideoLayout { Type = "bestFit" };
        public static OpenTokVideoLayout Pip => new OpenTokVideoLayout { Type = "pip" };
        public static OpenTokVideoLayout VerticalPresentation => new OpenTokVideoLayout { Type = "verticalPresentation" };
        public static OpenTokVideoLayout HorizontalPresentation => new OpenTokVideoLayout { Type = "horizontalPresentation" };

        public static OpenTokVideoLayout Custom(string css) => new OpenTokVideoLayout { Type = "custom", Stylesheet = css };
    }
}

File: OpenTokFs/RequestTypes/OpenTokVideoLayout.fs
namespace OpenTokFs.RequestTypes

open Newtonsoft.Json

type OpenTokVideoLayout() =
    member val ``type``: string = "" with get, set

    [<JsonProperty(NullValueHandling = NullValueHandling.Ignore)>]
    member val stylesheet: string = null with get, set

    static member BestFit = new OpenTokVideoLayout(``type`` = "bestFit")
    static member Pip = new OpenTokVideoLayout(``type`` = "pip")
    static member VerticalPresentation = new OpenTokVideoLayout(``type`` = "verticalPresentation")
    static member HorizontalPresentation = new OpenTokVideoLayout(``type`` = "horizontalPresentation")
    static member Custom css = new OpenTokVideoLayout(``type`` = "custom", stylesheet = css)
